### PR TITLE
StatementPositionRule autocorrect support

### PR DESCRIPTION
**Pull request by nicolai86**
This PR adds support to automatically correct the StatementPositionRule

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/StatementPositionRule.swift [hunk 1]
```diff
@@ -28,15 +28,59 @@ public struct StatementPositionRule: Rule {
             "}↓  else {",
             "}↓\ncatch {",
             "}\n\t↓  catch {"
+        ],
+        corrections: [
+          "}\n else {\n":"} else {\n",
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/StatementPositionRule.swift]: elsewhere in the code, we use a single space after the colon in dictionaries: `"{\n": "}"`

Comment (jpsim) [inline Source/SwiftLintFramework/Rules/StatementPositionRule.swift]: we also use 4-space indentation. please update all your changes in this file to match that

File: Source/SwiftLintFramework/Rules/StatementPositionRule.swift [hunk 2]
```diff
@@ -28,15 +28,59 @@ public struct StatementPositionRule: Rule {
...
-        let pattern = "((?:\\}|[\\s] |[\\n\\t\\r])\\b(?:else|catch))\\b"
+      let pattern = "(?:\\}|[\\s] |[\\n\\t\\r])\\b(?:else|catch)\\b"
+
+      return violationRangesInFile(file, withPattern: pattern).flatMap { range in
+        return StyleViolation(ruleDescription: self.dynamicType.description,
+          location: Location(file: file, characterOffset: range.location))
+      }
+    }
+
+    public func correctFile(file: File) -> [Correction] {
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/StatementPositionRule.swift]: This should be:

```swift
public func correctFile(file: File) -> [Correction] {
    return correctFile(file, withPattern: "\\}\\s+((?:else|catch))\\b")
}
```

No need to declare a `patterns()` function either.

Comment (jpsim) [inline Source/SwiftLintFramework/Rules/StatementPositionRule.swift]: Actually, one more thing: just move the body of `correctFile(_:withPattern:)` to this function.

**Comment by jpsim** [comment]
Thanks for the PR!

Don't forget to update CHANGLOG.md following the [contribution guidelines](https://github.com/realm/SwiftLint/blob/master/CONTRIBUTING.md).

**Comment by nicolai86** [comment]
@jpsim thanks for the review. I've adjusted the code accordingly. I'll update the CHANGELOG.md next. thanks

**Comment by nicolai86** [comment]
@jpsim the CHANGELOG.md is now updated as well.

**Comment by jpsim** [comment]
Alright, nice work @nicolai86! :clap:

**Comment by NachoSoto** [comment]
:tada: so excited to see SwiftLint getting better and better.

**Comment by jpsim** [comment]
Totally! I'm excited to get an increasing number of contributions from the community! :smile:

**Comment by nicolai86** [comment]
I'm really glad that contributing to SwiftLint is so easy. Thanks for reviewing it so quickly.

**Pull request merged**